src/shared/components/ui/Tabs/Tab.js

```diff
@@ -34,8 +44,22 @@ const Tab = ({ name, className, onClick, children, ...restProps }) => {
     onClick(event)
   }, [])
 
+  /**
+   * Handle disabled tab click
+   * invoke only custom handler without opening tab
+   * @type {function}
+   */
+  const handleDisabledTabClick = useCallback(
+    event => onDisabledClick(event),
+    []
+  )
+
   return (
-    <span className={classNames} onClick={handleClick} {...restProps}>
+    <span
+      className={classNames}
+      onClick={isDisabled ? handleDisabledTabClick : handleClick}
```

Comment:
nie widzę sensu robienia dodatkowego onClicka dla propsa disabled - chcę coś przekazać to chce, nie to nie. Każdy powinien sobie poradzić z tym czy coś tu przekazuje, czy nie

Reply:
możesz zostawić to w tej funkcji handleDisabledTabClick dla czytelności, ale użyj po prostu onClick

Reply:
Zrobiłem już If-a chciałem go uniknąć w poprzednim rozwiązaniu, no ale może być też. Jak ich nie ma więcej to jest ok